cognite/client/data_classes/relationships.py

```diff
@@ -1,139 +1,129 @@
+        confidence (float): Confidence value of the existence of this relationship. Generated relationships should provide a realistic score on the likelihood of the existence of the relationship. Relationships without a confidence value can be interpreted at the discretion of each project.
+        data_set_id (Dict[str, Any]): The id or externalId of the dataset this relationship belongs to.
+        labels (List[Label]): A list of the labels associated with this resource item.
         created_time (int): Time, in milliseconds since Jan. 1, 1970, when this relationship was created in CDF.
         last_updated_time (int): Time, in milliseconds since Jan. 1, 1970, when this relationship was last updated in CDF.
         cognite_client (CogniteClient): The client to associate with this object.
     """
 
     def __init__(
         self,
-        source: Dict[str, Any] = None,
-        target: Dict[str, Any] = None,
+        external_id: str = None,
+        source_external_id: str = None,
+        source_type: str = None,
+        target_external_id: str = None,
+        target_type: str = None,
         start_time: int = None,
         end_time: int = None,
         confidence: float = None,
-        data_set: str = None,
-        external_id: str = None,
-        relationship_type: str = None,
+        data_set_id: Dict[str, Any] = None,
+        labels: List[Label] = None,
         created_time: int = None,
         last_updated_time: int = None,
         cognite_client=None,
     ):
-        self.source = source
-        self.target = target
+        self.external_id = external_id
+        self.source_external_id = source_external_id
+        self.source_type = source_type
+        self.target_external_id = target_external_id
+        self.target_type = target_type
         self.start_time = start_time
         self.end_time = end_time
         self.confidence = confidence
-        self.data_set = data_set
-        self.external_id = external_id
-        self.relationship_type = relationship_type
+        self.data_set_id = data_set_id
         self.created_time = created_time
         self.last_updated_time = last_updated_time
+        self.labels = labels
         self._cognite_client = cognite_client
 
-    def _copy_resolve_targets(self):
+    def _validate_resource_types(self):
         rel = copy.copy(self)
-        rel.source = self._resolve_target(rel.source)
-        rel.target = self._resolve_target(rel.target)
+        self._validate_target(rel.source_type)
+        self._validate_target(rel.target_type)
         return rel
 
     @staticmethod
-    def _resolve_target(target):
+    def _validate_target(target):
         if isinstance(target, dict) or target is None:
             return target
```

Comment:
source/target type should be a string, not a dict, in the v1 implementation. I would remove this if statement